### PR TITLE
opencontrail: close col only if it was loaded

### PR DESCRIPTION
**Pull request by eonpatapon**
When collection.LoadCollection returns an error we should not call
Close() as col will refer to a nil pointer.

Rewrited the loop to make it more simple to read.

### CODE REVIEW

File: topology/probes/opencontrail.go [hunk 1]
```diff
@@ -100,27 +100,30 @@ func (mapper *OpenContrailProbe) retrieveMetadata(metadata graph.Metadata, itf c
...
 	for i := 0; i < try; i++ {
-		col, e := collection.LoadCollection(descriptions.Interface(), []string{fmt.Sprintf("%s:%d", host, port)})
-		err = e
-		if e == nil {
-			itf, e := col.SearchStrictUnique(name)
-			err = e
-			if e == nil {
-				return col, itf, e
-			}
+		if i > 0 {
```
Comment (safchain) [inline topology/probes/opencontrail.go]: nit: you could use common.Retry

Comment (eonpatapon) [reply topology/probes/opencontrail.go]: common.Retry expect a function that returns only an error, I don't see how I can use it here

Comment (safchain) [reply topology/probes/opencontrail.go]: That is not important ;) It seems to me you are implementing the exact same pattern. That could be something like this

```
fnc := func() error {
  col, err = collection.LoadCollection(descriptions.Interface(), []string{fmt.Sprintf("%s:%d", host, port)})
  if err != nil {
    return err
  }

  elem, err = col.SearchStrictUnique(name)
  if err != nil {
    col.Close()
    return err
  }

  return nil
}

common.Retry(fnc, 3, 500*timeMillisecond)
```

**Comment by safchain** [comment]
Some dependencies are missing

**Comment by eonpatapon** [comment]
run skydive-functional-tests-backend-orientdb

**Pull request merged**